### DISCUSSION TITLE
How to get hour from Offset?

### DISCUSSION THREAD

**Discussion by azzamsa** [BurntSushi/jiff#81] [Q&A] [answered]
Hi,

I wanted to port this Chrono code to Jiff equivalent.
Is there something like `offset().total(Unit::Hour)`?

```rust
fn main() {
    let offset_second = chrono::Local::now().offset().local_minus_utc();
    let offset_hour = chrono::Duration::seconds(offset_second.into()).num_hours();
    dbg!(&offset_hour);

    let offset_hour: jiff::tz::Offset = jiff::Zoned::now().offset();
    dbg!(&offset_hour);
}
````

Thanks for JIff! 🥞

**Comment by BurntSushi** [comment, marked as answer]
`offset.seconds() / (60 * 60)` should do the trick.

We can consider adding `offset.hours()` and `offset.minutes()` methods to make this easier. But I'm curious, what are you trying to do with the hours from an offset? (And specifically, this would ignore any minute or second component, since an offset isn't necessarily a whole number of hours.)

**Reply by azzamsa** [reply]
Wow, that's fast.

I also can do something like this.

```rust
fn main() {
    let offset = chrono::Local::now().offset().local_minus_utc();
    let offset = chrono::Duration::seconds(offset.into()).num_hours();
    dbg!(&offset);

    let offset = jiff::Zoned::now().offset().seconds();
    let offset = offset.seconds().total(Unit::Hour);
    dbg!(&offset);
}
```

I wanted to do calculation `(offset hours * n)` [here](https://github.com/azzamsa/islam/blob/d0ca199eaf1a91b884175b819c4045943527416c/src/salah/times.rs#L256)

Don't ask me where do I get the `n`. It is from a [paper](https://github.com/abougouffa/pyIslam/blob/master/docs/References.md#references-and-bibliography) :)
